spec/unit/pops/validator/validator_spec.rb

```diff
@@ -313,20 +327,14 @@ def validate(model)
       end
     end
 
-    it 'does not produce an error when function is at top level script' do
+    it 'does not produce an error when function is alone at top level script' do
       source = 'function y() {}'
       expect(validate(parse(source))).not_to have_issue(Puppet::Pops::Issues::NOT_ABSOLUTE_TOP_LEVEL)
     end
 
-    it 'does not produce an error when function is at top level file' do
-      source = 'function y() {}'
-
-      # We simulate a file by inserting a block between the program and the contained function
-      program = parse(source).current
-      function = program.body
-      program.body = Puppet::Pops::Model::BlockExpression.new
-      program.body.addStatements(function)
-      expect(validate(program)).not_to have_issue(Puppet::Pops::Issues::NOT_ABSOLUTE_TOP_LEVEL)
+    it 'does not produce an error when function is in a top level block' do
+      source = "function x() {}\nfunction y() {}"
+      expect(validate(parse(source))).not_to have_issue(Puppet::Pops::Issues::NOT_ABSOLUTE_TOP_LEVEL)
     end
```

Comment:
since some examples when it does not error where added, for completeness sake maybe also add one that checks that functions and other expressions can be mixed:

$a = 10
function foo() { }

etc